modules/service/src/main/resources/db/migration/V1044__etm_in_mode.sql

```diff
@@ -0,0 +1,343 @@
+-- Some observing modes, however, will have multiple science ETMs
+CREATE UNIQUE INDEX exposure_time_mode_role_constraint
+  ON t_exposure_time_mode (c_observation_id, c_role)
+  WHERE c_role IN (
+    'requirement'::e_exposure_time_mode_role,
+    'acquisition'::e_exposure_time_mode_role
+  );
+
+-- Populate the requirement ETMs from the existing requirements in t_observation
+INSERT INTO t_exposure_time_mode (
+  c_observation_id,
+  c_role,
+  c_exposure_time_mode,
+  c_signal_to_noise,
+  c_signal_to_noise_at,
+  c_exposure_time,
+  c_exposure_count
+)
+SELECT
+  c_observation_id,
+  'requirement'::e_exposure_time_mode_role,
+  c_exp_time_mode,
+  c_etm_signal_to_noise,
+  c_etm_signal_to_noise_at,
+  c_etm_exp_time,
+  c_etm_exp_count
+FROM t_observation
+WHERE c_exp_time_mode IS NOT NULL;
+
+DROP VIEW IF EXISTS v_observation;
+DROP VIEW IF EXISTS v_generator_params;
+
+-- Drop the now redundant columns in t_observation
+ALTER TABLE t_observation
+  DROP COLUMN c_exp_time_mode,
+  DROP COLUMN c_etm_signal_to_noise,
+  DROP COLUMN c_etm_signal_to_noise_at,
+  DROP COLUMN c_etm_exp_time,
+  DROP COLUMN c_etm_exp_count;
+
+-- Migrate the existing long slit observing mode tables to have acquisition and
+-- science ETMs based on a default (for acq) and the requirement ETM (for sci).
+DO $$
+DECLARE
+  mode TEXT[];
+  modes CONSTANT TEXT[][] := ARRAY[
+    ['Flamingos2 Long Slit',
+     't_flamingos_2_long_slit',
+     'f.c_wavelength',
+     'LEFT JOIN t_f2_filter f ON f.c_tag = m.c_filter'
+    ],
+    ['GMOS North Long Slit',
+     't_gmos_north_long_slit',
+     'm.c_central_wavelength',
+     ''
+    ],
+    ['GMOS South Long Slit',
+     't_gmos_south_long_slit',
+     'm.c_central_wavelength',
+     ''
+    ]
+  ];
+BEGIN
+  FOREACH mode SLICE 1 IN ARRAY modes
+  LOOP
+    RAISE NOTICE 'Converting mode %', mode[1];
+
+    -- Copy requirement ETM into science ETM for every observation with this
+    -- observing mode.
+    EXECUTE format($sql$
+      INSERT INTO t_exposure_time_mode (
+        c_observation_id,
+        c_role,
+        c_exposure_time_mode,
+        c_signal_to_noise,
+        c_signal_to_noise_at,
+        c_exposure_time,
+        c_exposure_count
+      )
+      SELECT
+        m.c_observation_id,
+        'science'::e_exposure_time_mode_role,
+        e.c_exposure_time_mode,
+        e.c_signal_to_noise,
+        e.c_signal_to_noise_at,
+        e.c_exposure_time,
+        e.c_exposure_count
+      FROM %I m
+      INNER JOIN t_exposure_time_mode e
+        ON e.c_observation_id = m.c_observation_id
+       AND e.c_role = 'requirement';
+    $sql$, mode[2]);
+
+    -- Remove observing mode entries without a corresponding science ETM.  That
+    -- is, those that never had a requirement exposure time mode.  In order to
+    -- actually execute an observation, a requirement exposure time mode must
+    -- have been present. A science ETM is now going to be required.
+    EXECUTE format($sql$
```

Comment:
will this delete observations? what happens if you started to do a proposal and left it in between? do we need to drop other related data? I'm not sure but maybe some attachments? targets, conditions, etc